samples/WOCCatalog/WOCCatalog/CoreMotionViewController.mm

```diff
@@ -254,4 +267,100 @@ -(void)gyroStartButtonPressed:(UIButton*)button {
     }
 }
 
+
+// Methods for Magnetometer
+-(void)setupMagnetometer {
+    magnetoLabel = [[UILabel alloc] initWithFrame:CGRectMake(10, magnetoHeight - 5, 150, 50)];
+    [magnetoLabel setBackgroundColor:[UIColor whiteColor]];
+    [magnetoLabel setText:@"Magnetometer"];
+    [magnetoLabel setTextAlignment:NSTextAlignmentLeft];
+    [scrollView addSubview:magnetoLabel];
+
+    magnetoStartButton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
+    [magnetoStartButton setTitle:@"Start" forState:UIControlStateNormal];
+    magnetoStartButton.frame = CGRectMake(100, magnetoHeight + 50, buttonLength, 40);
+    [magnetoStartButton addTarget:self action:@selector(magnetoStartButtonPressed:) forControlEvents:UIControlEventTouchUpInside];
+    [scrollView addSubview:magnetoStartButton];
+
+    magnetoStopButton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
+    [magnetoStopButton setTitle:@"Stop" forState:UIControlStateNormal];
+    magnetoStopButton.frame = CGRectMake(180, magnetoHeight + 50, buttonLength, 40);
+    magnetoStopButton.enabled = NO;
+    [magnetoStopButton addTarget:self action:@selector(magnetoStopButtonPressed:) forControlEvents:UIControlEventTouchUpInside];
+    [scrollView addSubview:magnetoStopButton];
+
+    NSArray* segmentTextContent = [NSArray arrayWithObjects:@"with Handler", @"without Handler", nil];
+    magnetoHandlerSegment = [[UISegmentedControl alloc] initWithItems:segmentTextContent];
+    magnetoHandlerSegment.frame = CGRectMake(130, magnetoHeight, buttonLength * 3 + 20, 40);
+    [magnetoHandlerSegment addTarget:self action:@selector(magnetoStopButtonPressed:) forControlEvents:UIControlEventValueChanged];
+    magnetoHandlerSegment.selectedSegmentIndex = 0;
+    [scrollView addSubview:magnetoHandlerSegment];
+
+    magnetoUpdateButton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
+    [magnetoUpdateButton setTitle:@"Poll Values" forState:UIControlStateNormal];
+    magnetoUpdateButton.frame = CGRectMake(270, magnetoHeight + 50, buttonLength * 1.3f, 40);
+    [magnetoUpdateButton addTarget:self action:@selector(magnetoUpdateButtonPressed:) forControlEvents:UIControlEventTouchUpInside];
+    magnetoUpdateButton.enabled = NO;
+    [scrollView addSubview:magnetoUpdateButton];
+
+    magnetoVal = [[UILabel alloc] initWithFrame:CGRectMake(0, magnetoHeight + 100, 350, 40)];
+    [magnetoVal setBackgroundColor:[UIColor whiteColor]];
+    [magnetoVal setText:@"X: 0.000        Y: 0.000        Z: 0.000"];
+    [magnetoVal setTextAlignment:NSTextAlignmentRight];
+    [scrollView addSubview:magnetoVal];
+}
+
+
+-(void)magnetoStopUpdates {
+    dispatch_queue_t backgroundQueue = dispatch_queue_create("stopMagneto", NULL);
```

Comment:
Why do we stop updates from background queue but start updates from main queue?  Also, if you really need to use background queue here, please use: dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0) instead of creating a new dispatch queue.